operations/fetch.go

```diff
@@ -194,18 +223,25 @@ func fetchSource(ctx context.Context, ac, rc *legacyClient, comm client.Communic
 }
 
 type cloneOptions struct {
-	owner      string
-	repository string
-	revision   string
-	rootDir    string
-	branch     string
-	token      string
-	depth      uint
+	owner        string
+	repository   string
+	revision     string
+	rootDir      string
+	branch       string
+	token        string
+	depth        uint
+	is_app_token bool
 }
 
 func clone(opts cloneOptions) error {
+	var cloneArgs []string
 	// clone the repo first
-	cloneArgs := []string{"clone", thirdparty.FormGitURL("github.com", opts.owner, opts.repository, opts.token)}
+	if opts.is_app_token {
+		cloneArgs = []string{"clone", thirdparty.FormGitURLForApp("github.com", opts.owner, opts.repository, opts.token)}
+	} else {
+		cloneArgs = []string{"clone", thirdparty.FormGitURL("github.com", opts.owner, opts.repository, opts.token)}
+	}
```

Comment:
Rather than having to pass around more information in the CLI and in the agent about whether it's an OAuth token vs. an installation token, could we instead use the token's prefix to determine the clone URL? Feel like that would eliminate the need to pass around the token type. [This link](https://github.blog/engineering/platform-security/behind-githubs-new-authentication-token-formats/#identifiable-prefixes) suggests that they always use `ghp_` for personal access tokens and `ghs_` for installation tokens, so we could use that prefix to decide between `https://github.com:x-oauth-basic@...` and `https://github.com:x-access-token@`, rather than us having to manually pass in CLI options or figure out the token type in the agent.

Reply:
I've shied away from that because as far as I know, they don't really provide a guarantee to not change this format, making it safer for the long run to keep track of it ourselves.